aten/src/ATen/cpu/vec256/vec256_float_neon.h

```diff
@@ -642,6 +642,853 @@ Vec256<float> inline fmadd(const Vec256<float>& a, const Vec256<float>& b, const
+      volatile register float32x4_t high asm("q4");
+      __asm__ __volatile__ (
+          "vld1.32 {d6, d7, d8, d9}, [%[in_ptr]]\n\t"
+          : "=w" (low), "=w" (high)
+          : [in_ptr] "r" (ptr)
+          :);
+      return Vec256<float>(low, high);
+    }
+    else if (count == (size() >> 1)) {
+      volatile register float32x4_t low asm("q3");
+      float32x4_t high{0.f, 0.f, 0.f, 0.f};
+      Vec256<float> res;
+      __asm__  __volatile__ (
+          "vld1.32 {d6, d7}, [%[in_ptr]]\n\t"
+          : "=w" (low)
+          : [in_ptr] "r" (ptr)
+          :);
+      return Vec256<float>(low, high);
+    }
+    else {
+      __at_align32__ float tmp_values[size()];
+      for (auto i = 0; i < size(); ++i) {
+        tmp_values[i] = 0.0;
+      }
+      std::memcpy(
+          tmp_values,
+          reinterpret_cast<const float*>(ptr),
+          count * sizeof(float));
+      volatile register float32x4_t low asm("q3");
+      volatile register float32x4_t high asm("q4");
+      __asm__  __volatile__ (
+          "vld1.32 {d6, d7, d8, d9}, [%[in_ptr]]\n\t"
+          : "=w" (low), "=w" (high)
+          : [in_ptr] "r" (tmp_values)
+          :);
+      return Vec256<float>(low, high);
+    }
+    return Vec256<float>();
+  }
+  void store(void* ptr, int64_t count = size()) const {
+    if (count == size()) {
+      __asm__  __volatile__ (
+          "vmov.f32 q3, %[in_low]\n\t"
+          "vmov.f32 q4, %[in_high]\n\t"
+          "vst1.32 {d6, d7, d8, d9}, [%[in_ptr]]\n\t"
+          :
+          : [in_ptr] "r" (ptr),
+            [in_low] "w" (values.val[0]), [in_high] "w" (values.val[1])
+          : "q3", "q4");
+    }
+    else if (count == (size() >> 1)) {
+      __asm__  __volatile__ (
+          "vmov.f32 q3, %[in_low]\n\t"
+          "vst1.32 {d6, d7}, [%[in_ptr]]\n\t"
+          :
+          : [in_ptr] "r" (ptr), [in_low] "w" (values.val[0])
+          : "q3");
+    }
+    else {
+      float tmp_values[size()];
+      __asm__  __volatile__ (
+          "vmov.f32 q3, %[in_low]\n\t"
+          "vmov.f32 q4, %[in_high]\n\t"
+          "vst1.32 {d6, d7, d8, d9}, [%[in_ptr]]\n\t"
+          :
+          : [in_ptr] "r" (tmp_values),
+            [in_low] "w" (values.val[0]), [in_high] "w" (values.val[1])
+          : "q3", "q4");
+      std::memcpy(ptr, tmp_values, count * sizeof(float));
+    }
+  }
+  inline const float32x4_t& get_low() const {
+    return values.val[0];
+  }
+  inline float32x4_t& get_low() {
+    return values.val[0];
+  }
+  inline const float32x4_t& get_high() const {
+    return values.val[1];
+  }
+  inline float32x4_t& get_high() {
+    return values.val[1];
+  }
+  // Very slow implementation of indexing.
+  // Only required because vec256_qint refers to this.
+  // Once we specialize that implementation for ARM
+  // this should be removed. TODO (kimishpatel)
+  const float operator[](int idx) const {
+    __at_align32__ float tmp[size()];
+    store(tmp);
```

Comment:
We have a store() and a loadu().  You can consider adding aligned versions of these functions for internal use in this class, and then use them where you have control over the alignment of data like here.  Just a suggestion.

Reply:
I dont know for sure if it makes perf difference. It probably avoids address alignment check, but my understanding is that if the address is aligned there is no perf penalty but if it is not then there is one cycle penalty. In x86 if you use aligned vs of load with unaligned address you get exception, not sure if thats the case here but either way it not known to me if this introduces performance penalty.